src/room.rs

```diff
@@ -1808,6 +1808,20 @@ impl PeerEventHandler for InnerRoom {
             }
         }
 
+        if let Some(peer_state) = self.peers.state().get(peer_id) {
+            peer_state
+                .get_tracks()
+                .into_iter()
+                .flat_map(|track_id| {
+                    self.connections
+                        .track_connections(&track_id)
+                        .unwrap_or_default()
```

Comment:
We do not need extra allocation of `Vec` here. Just `.filter_map().flatten()` and that's it.